tracer/src/Datadog.Trace/OTelMetrics/MetricStreamIdentity.cs

```diff
@@ -21,6 +24,19 @@ public MetricStreamIdentity(Instrument instrument, InstrumentType instrumentType
         {
             InstrumentName = instrument.Name;
             MeterName = instrument.Meter.Name;
+            MeterVersion = instrument.Meter.Version ?? string.Empty;
+
+            // Duck typing works at runtime - checks if Tags property exists (.NET 8+)
+            try
+            {
+                var meterDuck = instrument.Meter.DuckCast<IMeterDuck>();
+                MeterTags = meterDuck.Tags?.ToArray() ?? [];
+            }
+            catch
+            {
+                MeterTags = [];
+            }
```

Comment:
I'd rather we not rely on throwing an exception here (in fact I don't actually know if it does throw 🤔 )

Could we swap to `TryDuckCast` or `DuckAs`?

Reply:
Done in my last commit: